client/lib/paths/login/index.js

```diff
@@ -51,5 +52,9 @@ export function login( {
 		url = addQueryArgs( { email_address: emailAddress }, url );
 	}
 
+	if ( oauth2ClientId ) {
```

Comment:
Minor but still worth it imo: we could check that `oauth2ClientId` is a number here.